src/common/snippets/include/snippets/lowered/specific_loop_iter_handlers.hpp

```diff
@@ -13,41 +14,41 @@ namespace lowered {
 
 class SpecificIterationHandlers {
 public:
-    enum class HandlerType { FIRST_ITER, MAIN_BODY, LAST_ITER };
     SpecificIterationHandlers() = default;
     SpecificIterationHandlers(size_t loop_work_amount, size_t loop_increment);
     SpecificIterationHandlers(pass::PassPipeline first_iter_handlers,
                               pass::PassPipeline main_body_handlers,
                               pass::PassPipeline last_iter_handlers);
 
-    const pass::PassPipeline& get_first_iter_handlers() const;
-    const pass::PassPipeline& get_main_iter_handlers() const;
-    const pass::PassPipeline& get_last_iter_handlers() const;
-    static SpecificIterationHandlers merge_handlers(const SpecificIterationHandlers& lhs, const SpecificIterationHandlers& rhs);
-
-    template <HandlerType Type,
-              typename T,
-              class... Args,
-              typename std::enable_if<Type == HandlerType::FIRST_ITER, bool>::type = true>
-    void register_handler(Args&&... args) {
-        m_first_iter_handlers.register_pass<T>(args...);
+    template <SpecificLoopIterType Type>
+    const pass::PassPipeline& get_passes() const {
+        switch (Type) {
+            case SpecificLoopIterType::FIRST_ITER:
+                return m_first_iter_handlers;
+            case SpecificLoopIterType::MAIN_BODY:
+                return m_main_body_handlers;
+            case SpecificLoopIterType::LAST_ITER:
+                return m_last_iter_handlers;
+            default:
+                OPENVINO_THROW("Unknown SpecificLoopIterType");
+        }
     }
 
-    template <HandlerType Type,
-              typename T,
-              class... Args,
-              typename std::enable_if<Type == HandlerType::MAIN_BODY, bool>::type = true>
-    void register_handler(Args&&... args) {
-        m_main_body_handlers.register_pass<T>(args...);
+    template <SpecificLoopIterType Type, typename Pass, class... Args>
+    void register_pass(Args&&... args) {
+        switch (Type) {
+            case SpecificLoopIterType::FIRST_ITER:
+                return m_first_iter_handlers.register_pass<Pass>(args...);
+            case SpecificLoopIterType::MAIN_BODY:
+                return m_main_body_handlers.register_pass<Pass>(args...);
+            case SpecificLoopIterType::LAST_ITER:
+                return m_last_iter_handlers.register_pass<Pass>(args...);
+            default:
+                OPENVINO_THROW("Unknown SpecificLoopIterType");
+        }
```

Comment:
I hope I didn't hurt your feelings as author of this impl 😄 
But for me it's more convenient when only one method contains all possible template arguments because if I decide to add new `type`, I just can add new `case()` here.
Also it takes up less code space))

Reply:
As I said, I was just wondering :) I am satisfied your explanation, thank you! 😄